Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp

```diff
@@ -295,29 +341,53 @@ void MemoryViewWidget::Update()
       });
       break;
     }
-
-    if (row_breakpoint)
-    {
-      bp_item->setData(Qt::DecorationRole, Resources::GetScaledThemeIcon("debugger_breakpoint")
-                                               .pixmap(QSize(rowHeight(0) - 3, rowHeight(0) - 3)));
-    }
   }
+}
 
-  setColumnWidth(0, rowHeight(0));
+void MemoryViewWidget::UpdateBreakpointTags()
+{
+  if (Core::GetState() != Core::State::Paused)
+    return;
 
-  // Number of characters possible.
-  int max_length = GetCharacterCount(m_type);
+  for (int i = 0; i < rowCount(); i++)
+  {
+    bool row_breakpoint = true;
```

Comment:
Not really your fault, but this is a bit broken, eg. set a row breakpoint on 16 bytes per row, then switch to 4 bytes per row and try to unset them. I'd just remove the row breakpoint altogether and just keep the red backdrop and right-click menu breakpoints, but ymmv.

Reply:
Hmm I could set/unset multiple breakpoints according to row length + type size.